src/lib/InstantSearch.ts

```diff
@@ -64,24 +95,24 @@ class InstantSearch extends EventEmitter {
       throw new Error(withUsage('The `searchClient` option is required.'));
     }
 
-    if (typeof options.urlSync !== 'undefined') {
+    if (typeof (options as any).urlSync !== 'undefined') {
       throw new Error(
         withUsage(
           'The `urlSync` option was removed in InstantSearch.js 3. You may want to use the `routing` option.'
         )
       );
     }
 
-    if (typeof searchClient.search !== 'function') {
+    if (typeof (searchClient as any).search !== 'function') {
```

Comment:
`searchClient` should always be typed with a `search` method. Why do you need this cast?

Reply:
TS would complain about a needless comparison if search is always defined